packages/notifi-react/lib/context/NotifiHistoryContext.tsx

```diff
@@ -75,12 +82,25 @@ export const NotifiHistoryContextProvider: FC<
             includeRead: isIncludeRead,
           })
           .then((res) => {
-            const historyItemIdMap = new Map(
+            const existingItemIdMap = new Map(
               historyItems.map((item) => [item.id, item]),
             );
+
+            const fusionEventIdMap = new Map(
+              fusionEventTopics.map((topic) => [
+                topic.fusionEventDescriptor.id,
+                topic,
+              ]),
+            );
+
             const newItems = res?.nodes
               ?.map(parseHistoryItem)
-              .filter((item) => !historyItemIdMap.has(item.id));
+              .filter(
+                (item) =>
+                  !existingItemIdMap.has(item.id) &&
+                  fusionEventIdMap.has(item.fusionEventId),
```

Comment:
We use the `historyEntry.NotifiData.EventTypeId` to filter the corresponding history against the particular card.
> **TBC**: We will need to ensure the backward compatibility, meaning all `historyItemEntry`s have `.NotifiData.EventTypeId` property so that it will not break the existing tenents' users UX. CC: @nimesh-notifi